src/Type/IterableIterableType.php

```diff
@@ -135,6 +135,16 @@ public function getIterableValueType(): Type
 		return $this->getItemType();
 	}
 
+	public function isOffsetAccesible(): TrinaryLogic
+	{
+		return TrinaryLogic::createMaybe();
+	}
+
+	public function getOffsetValueType(): Type
+	{
+		return $this->getIterableValueType();
```

Comment:
This is a nice hack, but it's wrong. You should return `mixed` type.

Reply:
But in your case result from `returnArrayObjectOfStrings()` (see test) will be always mixed.

Reply:
Yes, that is correct. Iteration is not the same operation as array access. To support `returnArrayObjectOfStrings()` you would need generic version of `ArrayObject` class.

Reply:
Or at least generic version of `ArrayAccess` interface.